system_metrics_collector/src/topic_statistics_collector/received_message_age.hpp

```diff
@@ -18,15 +18,43 @@
 #include <chrono>
 #include <string>
 #include <sstream>
+#include <type_traits>
 
 #include "topic_statistics_collector.hpp"
 
-#include "message_filters/message_traits.h"
 #include "rcl/time.h"
-#include "rclcpp/clock.hpp"
 #include "rcutils/logging_macros.h"
 
 
+namespace
+{
+template<typename M, typename = void>
+struct HasHeader : public std::false_type {};
+
+template<typename M>
+struct HasHeader<M, decltype((void) M::header)>: std::true_type {};
+
+template<typename M, typename Enable = void>
+struct TimeStamp
+{
+  static int64_t value(const M & m)
```

Comment:
```suggestion
  static int64_t value(const M &)
```

and remove `(void)m`.

Reply:
That's neat.